### DISCUSSION TITLE
Resizing Tables and Panels

### DISCUSSION THREAD

**Discussion by andtii** [spectreconsole/spectre.console#356] [Q&A]
If you have a pretty wide table and the console window is smaller then the table it renders nicely like this 
![image](https://user-images.githubusercontent.com/1434571/114045601-6ad12e00-9888-11eb-90b5-fd659cc2b6d4.png)

But when i resize the console window to fit the content the whole table gets destroyed like this
![image](https://user-images.githubusercontent.com/1434571/114045773-93f1be80-9888-11eb-875d-ffc374838409.png)

Is there a way to redraw the table or is there something i have missed? 

`
 AnsiConsole.Render(new Table()
                .Border(TableBorder.DoubleEdge)
                .Title("TABLE [yellow]TITLE[/]")
                .Caption("TABLE [yellow]CAPTION[/]")
                .AddColumn(new TableColumn(new Panel("[u]ABC[/]").BorderColor(Color.Red)).Footer("[u]FOOTER 1[/]"))
                .AddColumn(new TableColumn(new Panel("[u]DEF[/]").BorderColor(Color.Green)).Footer("[u]FOOTER 2[/]"))
                .AddColumn(new TableColumn(new Panel("[u]GHI[/]").BorderColor(Color.Blue)).Footer("[u]FOOTER 3[/]"))
                .AddColumn(new TableColumn(new Panel("[u]GHI[/]").BorderColor(Color.Blue)).Footer("[u]FOOTER 3[/]"))
                .AddRow(
                        new Text("747c5abe-3d97-49eb-8f36-4e81f8edd401"),
                        new Text("747c5abe-3d97-49eb-8f36-4e81f8edd401"),
                        new Text("747c5abe-3d97-49eb-8f36-4e81f8edd401"),
                        new Text("747c5abe-3d97-49eb-8f36-4e81f8edd401"))
                );
`

BTW thnx for this awesome library!

**Comment by phil-scott-78** [comment]
I'm afraid that's just how consoles handle output. If we want to output a full width table, for example, we can detect the screen is 90 characters wide the title is 10. So the console "we want 40 = characters then the heading and 40 more = characters". Console renders and it looks fine.  But if you resize the window the console doesn't know our intentions when we outputted the characters, so it just tries to fit everything on the screen. 

In the future we might have support for custom widgets that can be constantly be redrawn which would allow us to react to screensize changes in between rendering as long as the app was in its render loop.
